app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -65,21 +69,24 @@ class KiwixMainActivity : CoreMainActivity() {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_kiwix_main)
 
-    navController = findNavController(R.id.nav_host_fragment)
     navController.addOnDestinationChangedListener(finishActionModeOnDestinationChange)
-    appBarConfiguration = AppBarConfiguration(
-      setOf(
-        R.id.navigation_downloads,
-        R.id.navigation_library,
-        R.id.navigation_reader
-      ), navigation_container
-    )
     drawer_nav_view.setupWithNavController(navController)
     drawer_nav_view.setNavigationItemSelectedListener { item ->
       closeNavigationDrawer()
       onNavigationItemSelected(item)
     }
     bottom_nav_view.setupWithNavController(navController)
+
+    navController.addOnDestinationChangedListener { _, destination, _ ->
+      if (destination.id == R.id.navigation_downloads ||
+        destination.id == R.id.navigation_library ||
+        destination.id == R.id.navigation_reader
+      ) {
+        bottom_nav_view.visibility = View.VISIBLE
```

Comment:
this is a good example of how a business rule is getting put in in a somewhat hidden way.

You started from this rule but then "made it code-y" instead of maintaining the sentence

> Show the bottom nav view when at a top level destination

```
val topLevelDestinations = setOf(R.id.etc, ...)

...

bottom_nav_view.visibility = if (destination.id in topLevelDestinations) Visible else GONE
```

also in kotlin I think all views have the `.isVisible` property so we get even better

```
bottom_nav_view.isVisible = if (destination.id in topLevelDestinations)
```

Reply:
What you are saying is that, your nice concise way is truer to the rule explained as a sentence. Which makes it easy to understand and read. And the code-y way is not as readable/true to the sentence? Just clarifying.

Reply:
yeah, in the previous version it requires knowledge to know that those ids are the top level destinations, it requires a mental leap to create the association, not a large one mind you but one the reader doesn't have to do.

the condensing to 1 line also helps a lot because 99 times out of a 100

```
if([expression])
[identicalBit] =[different bit 1]
else
[identicalBit] =[different bit 2]
```
is going to be much easier to  read as a ternary/if expression

Reply:
And really this would be a lot better if it could never break, if we ever add a top level destination you gotta update it here, which sucks.

In this kind of scenario enums usually come to the rescue because you could declare.

```
enum class TopLevelDestinations(val id:Int){
Reader(R.id.etc), Library(...
}
```

and our previous set gets replaced with `TopLevelDestinations.values` but we would then want to make the enums the source of truth about top level destinations and the graph is really what does that, bloody xml. So probably not worth it and we just have to keep updating this code on a change

Reply:
Alright, thank you for clarifying again. A conditional checking if the id is a top level destination feels like a common use case, it's odd that they don't have a way of tying that together.